python-avd/pyavd/_eos_designs/structured_config/network_services/vlan_interfaces.py

```diff
@@ -131,66 +137,70 @@ def _check_virtual_router_mac_address(vlan_interface_config: dict, variables: li
 
             _check_virtual_router_mac_address(vlan_interface_config, ["ipv6_address_virtuals"])
 
             if vlan_interface_config.get("ipv6_address_virtuals"):
                 # If any anycast IPs are set, we also enable link-local IPv6 per best practice, unless specifically disabled with 'ipv6_enable: false'
                 vlan_interface_config["ipv6_enable"] = get(vlan_interface_config, "ipv6_enable", default=True)
 
-        if vrf["name"] != "default":
-            vlan_interface_config["vrf"] = vrf["name"]
+        if vrf.name != "default":
+            vlan_interface_config["vrf"] = vrf.name
 
         # Adding IP helpers and OSPF via a common function also used for subinterfaces when uplink_type: lan
         self.shared_utils.get_additional_svi_config(vlan_interface_config, svi, vrf)
 
         return strip_empties_from_dict(vlan_interface_config)
 
-    def _get_vlan_interface_config_for_mlag_peering(self: AvdStructuredConfigNetworkServices, vrf: dict, vlan_id: int) -> dict:
+    def _get_vlan_interface_config_for_mlag_peering(
+        self: AvdStructuredConfigNetworkServices, vrf: EosDesigns._DynamicKeys.DynamicNetworkServicesItem.NetworkServicesItem.VrfsItem, vlan_id: int
+    ) -> dict:
         """Build full config for MLAG peering SVI for the given VRF."""
         vlan_interface_config = {
-            "tenant": vrf["tenant"],
+            "tenant": vrf._tenant,
             "type": "underlay_peering",
             "shutdown": False,
             "description": self.shared_utils.interface_descriptions.mlag_peer_l3_vrf_svi(
-                InterfaceDescriptionData(shared_utils=self.shared_utils, interface=f"Vlan{vlan_id}", vrf=vrf["name"], vlan=vlan_id)
+                InterfaceDescriptionData(shared_utils=self.shared_utils, interface=f"Vlan{vlan_id}", vrf=vrf.name, vlan=vlan_id)
             ),
-            "vrf": vrf["name"],
+            "vrf": vrf.name,
             "mtu": self.shared_utils.p2p_uplinks_mtu,
         }
         vlan_interface_config.update(self._get_vlan_ip_config_for_mlag_peering(vrf))
         return vlan_interface_config
 
-    def _get_vlan_ip_config_for_mlag_peering(self: AvdStructuredConfigNetworkServices, vrf: dict) -> dict:
+    def _get_vlan_ip_config_for_mlag_peering(
+        self: AvdStructuredConfigNetworkServices, vrf: EosDesigns._DynamicKeys.DynamicNetworkServicesItem.NetworkServicesItem.VrfsItem
+    ) -> dict:
         """
         Build IP config for MLAG peering SVI for the given VRF.
 
         Called from _get_vlan_interface_config_for_mlag_peering and prefix_lists.
         """
-        if self.shared_utils.underlay_rfc5549 and self.shared_utils.overlay_mlag_rfc5549:
+        if self.inputs.underlay_rfc5549 and self.inputs.overlay_mlag_rfc5549:
             return {"ipv6_enable": True}
 
-        if (mlag_ibgp_peering_ipv4_pool := vrf.get("mlag_ibgp_peering_ipv4_pool")) is not None:
+        if (mlag_ibgp_peering_ipv4_pool := vrf.mlag_ibgp_peering_ipv4_pool) is not None:
```

Comment:
Feels like we can remove the walrus operator alltogther here 

```suggestion
        if vrf.mlag_ibgp_peering_ipv4_pool:
```

and then replace the two usage below with this

Reply:
done